index.js

```diff
@@ -936,6 +936,25 @@ Connection.prototype._apiAuthRequest = function(opts, callback) {
   return resolver.promise;
 };
 
+function findValueForKeyInObject(obj, key, depth, maxDepth = 5) {
```

Comment:
Recursive function that goes till a pre-determined depth to find a key and return the value (or `undefined` if it can't find it in time).